model/user.js

```diff
@@ -1,15 +1,24 @@
 const mongoose = require("mongoose");
 const parameters = require("../utils/parameters.js");
 
+
+
 const UserSchema = mongoose.Schema({
     facebook: {
         id: String,
         token: String,
         email: String,
         name: String
-    }
+    },
     isAdmin : { type : Boolean , default : false } ,
-    accountBalance : Number ,
+    accountBalance : { type : Number , default : parameters.defaultbal},
+    stockShorted :[{
+        TotalPrice : Number,
+        TotalStock : Number
+    }],
+    stockHolding :[{
+        quantity : Number
+    }],
     activity : [{
     	company : [{
     		type : mongoose.Schema.Types.ObjectId ,
```

Comment:
why is company key an array? one entry of activity will belong to one company only right?

Reply:
so that it is easier to use findbyid method when searching for company

Reply:
I get that, but why is it an array? only a single company entry needs to go in the activity i think. If I'm buying x shares from a company, then that's one activity log to be pushed in the activity array, containing my transaction with only one company.

Reply:
my bad i wondered why it was working i will fix it